application/aspnet/sdk-example/example/Pages/Index.cshtml.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.Extensions.Logging;
+namespace example.Pages
+{
+    public class IndexModel : PageModel
+    {
+        private readonly ILogger<IndexModel> _logger;
+        public string Message { get; set; }
+        public IndexModel(ILogger<IndexModel> logger)
+        {
+            _logger = logger;
+        }
+        public void OnGet()
+        {
+            Message = response();
+        }
+        public static string response()
+        {
+            try
+            {
+                var client = new SecretHub.Client();
+                try
+                {
+                    var user = System.Environment.GetEnvironmentVariable("SECRETHUB_USERNAME");
+                    try
+                    {
+                        var username = client.Read(user + "/demo/username").Data;
+                        try
+                        {
+                            var password = client.Read(user + "/demo/password").Data;
+                            return "Hello, " + username + "!";
+                        }
+                        catch
+                        {
+                            return "error in reading demo password";
+                        }
+                    }
+                    catch
+                    {
+                        return "error in reading demo username";
+                    }
+                }
+                catch
+                {
+                    return "error in fetching username";
+                }
+            }
+            catch
+            {
+                return "error in client creation";

```

Comment:
Also we should log the message of the exception we catch, to help the user diagnose any issues they encounter.

Reply:
Let me know if this is more to your liking :D

Reply:
Yeah, it looks good to me :+1: